tests/semantic_analyser.cpp

```diff
@@ -124,6 +124,15 @@ TEST(semantic_analyser, builtin_variables)
   test("kprobe:f { probe }", 0);
   test("tracepoint:a:b { args }", 0);
   test("kprobe:f { fake }", 1);
+
+  auto bpftrace = get_mock_bpftrace();
+  test(*bpftrace, "i:ms:100 { printf(\"%d\\n\", cpid); }", 1, false);
+  test(*bpftrace, "i:ms:100 { @=cpid }", 1, false);
+  test(*bpftrace, "i:ms:100 { $a=cpid }", 1, false);
+  bpftrace->cmd_ = "sleep 1";
+  test(*bpftrace, "i:ms:100 { printf(\"%d\\n\", cpid); }", 0, false);
+  test(*bpftrace, "i:ms:100 { @=cpid }", 0, false);
+  test(*bpftrace, "i:ms:100 { $a=cpid }", 0, false);
```

Comment:
I think this deserves its own test case. The "builtin_variables" test is just meant to be a very simple check that the variables exist